### PR TITLE
gh-112535: Add test on _Py_ThreadId()

### PR DESCRIPTION
**Pull request by vstinner**
Add also test.support.Py_GIL_DISABLED constant.

* Issue: gh-112535

### CODE REVIEW

**Comment by vstinner** [comment]
> On FreeBSD x86-64, _Py_ThreadId() cannot be compared to Thread.ident or Thread.native_id:

Same on Linux ppc64le.

File: Modules/_testinternalcapi.c [hunk 1]
```diff
@@ -1625,6 +1625,17 @@ get_type_module_name(PyObject *self, PyObject *type)
 }
 
 
+#ifdef Py_GIL_DISABLED
+static PyObject *
+get_py_thread_id(PyObject *self, PyObject *type)
+{
+    uintptr_t tid = _Py_ThreadId();
+    Py_BUILD_ASSERT(sizeof(unsigned long long) >= sizeof(tid));
```
Comment (colesbury) [inline Modules/_testinternalcapi.c:1633]: You can just use C11 `static_assert()` now, which has better error messages.

**Review by colesbury — COMMENTED** [review]
I think having a test for `_PyThread_Id()` is good, but I I don't think we should bother comparing with `_thread.get_ident()`.

I think we want the following:
- `_PyThread_Id()` should be unique and non-zero
- `_PyThread_Id()` should not change for a given thread. For example, it should remain the same after a short sleep.
- It should be a multiple of four (i.e., at least 4-byte aligned). This isn't necessary for biased reference counting, but it makes implementing recursive mutexes easier and should be true for all our implementations of it.

File: Modules/_testinternalcapi.c (hunk 1 above)
Comment (vstinner) [reply Modules/_testinternalcapi.c:1633]: Py_BUILD_ASSERT() doesn't require an error message which makes my life easier, I don't have to write down an error message :-)

**Comment by vstinner** [comment]
> I think having a test for _PyThread_Id() is good, but I I don't think we should bother comparing with _thread.get_ident().

Ok, I removed this test.

> _PyThread_Id() should be unique and non-zero
> _PyThread_Id() should not change for a given thread. For example, it should remain the same after a short sleep.

I added tests for that.

> It should be a multiple of four (i.e., at least 4-byte aligned). This isn't necessary for biased reference counting, but it makes implementing recursive mutexes easier and should be true for all our implementations of it.

I will let you add a check for that later (in a following PR). I prefer to add tests step by step, especially if it's not strictly needed right now.

**Review by colesbury — APPROVED** [review]

File: Lib/test/test_importlib/test_windows.py [hunk 2]
```diff
@@ -112,7 +112,7 @@ def test_module_not_found(self):
 class WindowsExtensionSuffixTests:
     def test_tagged_suffix(self):
         suffixes = self.machinery.EXTENSION_SUFFIXES
-        abi_flags = "t" if sysconfig.get_config_var("Py_GIL_DISABLED") else ""
+        abi_flags = "t" if support.Py_GIL_DISABLED else ""
```
Comment (colesbury) [inline Lib/test/test_importlib/test_windows.py:115]: @vstinner  - looks like Windows tests are failing (possibly due to missing import for support:

https://github.com/python/cpython/actions/runs/7090430380/job/19297369118?pr=112709

File: Modules/_testinternalcapi.c [hunk 3]
```diff
@@ -1625,6 +1625,17 @@ get_type_module_name(PyObject *self, PyObject *type)
 }
 
 
+#ifdef Py_GIL_DISABLED
+static PyObject *
+get_py_thread_id(PyObject *self, PyObject *type)
```
Comment (corona10) [inline Modules/_testinternalcapi.c]:
```suggestion
get_py_thread_id(PyObject *self, PyObject *Py_UNUSED(ignored))
```

Comment (corona10) [reply Modules/_testinternalcapi.c]: Looks like type is not used at all :)

File: Lib/test/test_importlib/test_windows.py (hunk 2 above)
Comment (vstinner) [reply Lib/test/test_importlib/test_windows.py:115]: Ah right,  I saw it before. Thanks for the reminder. It should now be fixed.

**Pull request merged**